action.yml

```diff
@@ -20,17 +21,72 @@ inputs:
 branding:
   icon: "shield"
   color: "green"
+
 runs:
-  using: "docker"
-  image: "docker://ghcr.io/trufflesecurity/trufflehog:latest"
-  args:
-    - git
-    - file://${{ inputs.path }}
-    - --since-commit
-    - ${{ inputs.base }}
-    - --branch
-    - ${{ inputs.head }}
-    - --fail
-    - --no-update
-    - --github-actions
-    - ${{ inputs.extra_args }}
+  using: "composite"
+  steps:
+  - shell: bash
+    env:
+      REPO_PATH: ${{ inputs.path }}
+      BASE: ${{ inputs.base }}
+      HEAD: ${{ inputs.head }}
+      ARGS: ${{ inputs.extra_args }}
+    run: |
+      ##########################################
+      ## ADVANCED USAGE                       ##
+      ## Scan by BASE & HEAD user inputs      ##
+      ## If BASE == HEAD, exit with error     ##
+      ##########################################
+      if [ -n "$BASE" ] || [ -n "$HEAD" ]; then
+        if [ -n "$BASE" ]; then
+          base_commit=$(git rev-parse "$BASE" 2>/dev/null) || true
+        else
+          base_commit=""
+        fi
+        if [ -n "$HEAD" ]; then
+          head_commit=$(git rev-parse "$HEAD" 2>/dev/null) || true
+        else
+          head_commit=""
+        fi
+        if [ $base_commit == $head_commit ] ; then
```

Comment:
Definitely a valid use case. I'm open to adjusting this. I think we should ship this asap since some folks aren't scanning commits atm and think they are. But I'm down to think through what the behavior should look like.

Reply:
Separately, I'm reviewing this code before merging in and I'm looking at this line `if [ $base_commit == $head_commit ] ; then`. Do you think there are folks that leave `base` equal to `''` and `head` equal to `''`, so that they can trigger a full git repository history scan? @zricethezav